device.h

```diff
@@ -31,6 +36,22 @@ class Flashcart {
     const char* m_name;
     const size_t m_max_length;
 
+    // For devices that need to unlock flash and the like,
+    // and have issues reading when flash is unlocked.
+    virtual void setWriteState(bool state); // Default 'do-nothing' implementation provided.
```

Comment:
not sure if this requires a virtual method when it's only implemented per device?

Reply:
I don't remember how this stuff w/ C++ works, to be honest. Not really a fan of OO. So... I kinda just throw virtual on stuff when I think the class is gonna have to implement/override it.

Reply:
Oh, I see. when calling something on a "Flashcart" that should be invoking the method on the "AK2i", I want virtual. When it's always in the same class, e.g. a method that is already virtual calling one that isn't, then it isn't needed.

Reply:
In which case, yes, we want it to be virtual, because the global readFlash and writeFlash are in Flashcart class namespace, and thus will always resolve it to the do-nothing implementation that I provided.